ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/termsandconditions/v1/termsandconditions/clientlibs/site/js/termsandconditionsview.js

```diff
@@ -62,10 +62,12 @@
                     threshold: [1],
                 })
                 function onIntersection ([{isIntersecting}]) {
+                    const isDisabled = self._model._jsonModel?.enabled && !self._model._jsonModel?.readOnly;
                     if (isIntersecting) {
-                        self.children.filter(c => c.getModel()._jsonModel.fieldType === 'checkbox').forEach(cb => {
-                            cb.getModel().enabled = true;
-                        })
+                        if (isDisabled) {
+                              self.children.filter(c => c.getModel()._jsonModel.fieldType === 'checkbox').forEach(cb => {
+                                cb.getModel().enabled = true;
```

Comment:
Why is this set to enabled true when TnC is disabled

Reply:
Rishi this is the logic to enable checkbox after reader has completely scrolled down  the message if TNC component is => enabled = true and readOnly=false